<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scMerge: a package for merging scRNA-seq data • scMerge</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="scMerge: a package for merging scRNA-seq data">
<meta property="og:description" content="Remove batch effect and unwanted variations for single-cell RNA-seq expression matrix.
    The method includes the function of identification of stably expressed genes (SEG) from single-cell data, and also creating pseudo-replicates from the data.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scMerge</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/scMerge.html">Vignette</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Case Study
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Breast_Data/Breast_Data.html">Breast Cancer Data</a>
    </li>
    <li>
      <a href="articles/CellBench_Data/CellBench_Data.html">CellBench Data</a>
    </li>
    <li>
      <a href="articles/ESC_Data/ESC_Data.html">Embyronic Stem Cell Data</a>
    </li>
    <li>
      <a href="articles/Mouse_Liver_Data/Mouse_Liver_Data.html">Mouse Liver Data</a>
    </li>
    <li>
      <a href="articles/Neuronal_Data/Neuronal_Data.html">Neuronal Single Cell Data</a>
    </li>
    <li>
      <a href="articles/Pancreas4_Data/Pancreas4_Data.html">Pancreas 4 Data</a>
    </li>
    <li>
      <a href="articles/mESC_Data/mESC_Data.html">Mouse Embryonic Stem Cell Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/scMerge">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    <div id="scmerge" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#scmerge" class="anchor"></a>scMerge</h1></div>
<p><code>scMerge</code> is a R package for merging and normalising single-cell RNA-Seq datasets.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The installation process could take up to 5 minutes, depending if you have some of the packages pre-installed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Some CRAN packages required by scMerge</span>
<span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">"ruv"</span>, <span class="st">"rsvd"</span>, <span class="st">"igraph"</span>, <span class="st">"pdist"</span>, <span class="st">"proxy"</span>, <span class="st">"foreach"</span>, <span class="st">"doSNOW"</span>, <span class="st">"distr"</span>, <span class="st">"Rcpp"</span>, <span class="st">"RcppEigen"</span>))
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"theislab/kBET"</span>)

<span class="co"># Some BioConductor packages required by scMerge</span>
<span class="co"># try http:// if https:// URLs are not supported</span>
<span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="kw">c</span>(<span class="st">"SingleCellExperiment"</span>, <span class="st">"M3Drop"</span>))


<span class="co"># Installing scMerge and the data files using</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"SydneyBioX/scMerge.data"</span>)
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"SydneyBioX/scMerge"</span>)</code></pre></div>
</div>
<div id="vignette" class="section level2">
<h2 class="hasAnchor">
<a href="#vignette" class="anchor"></a>Vignette</h2>
<p>You can find the vignette at our website: <a href="https://sydneybiox.github.io/scMerge/index.html" class="uri">https://sydneybiox.github.io/scMerge/index.html</a>.</p>
</div>
<div id="case-studies" class="section level2">
<h2 class="hasAnchor">
<a href="#case-studies" class="anchor"></a>Case studies</h2>
<p>You can find a list of case studies here: <a href="https://sydneybiox.github.io/scMerge/articles/" class="uri">https://sydneybiox.github.io/scMerge/articles/</a>.</p>
</div>
<div id="contact-us" class="section level2">
<h2 class="hasAnchor">
<a href="#contact-us" class="anchor"></a>Contact us</h2>
<p>If you have any enquires, especially about performing <code>scMerge</code> integration on your data, then please contact <a href="mailto:bioinformatics@maths.usyd.edu.au">bioinformatics@maths.usyd.edu.au</a>.</p>
</div>
<div id="reference" class="section level2">
<h2 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h2>
<p><strong>scMerge: Integration of multiple single-cell transcriptomics datasets leveraging stable expression and pseudo-replication</strong></p>
<p>Yingxin Lin, Shila Ghazanfar, Kevin Y.X. Wang, Johann A. Gagnon-Bartsch, Kitty K. Lo, Xianbin Su, Ze-Guang Han, John T. Ormerod, Terence P. Speed, Pengyi Yang, Jean Y. H. Yang</p>
<p>doi: <a href="https://doi.org/10.1101/393280" class="uri">https://doi.org/10.1101/393280</a></p>
<p>BioRxiv preprint: <a href="https://www.biorxiv.org/content/early/2018/08/16/393280" class="uri">https://www.biorxiv.org/content/early/2018/08/16/393280</a></p>
</div>
</div>
  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/SydneyBioX/scMerge">https://​github.com/​SydneyBioX/​scMerge</a>
</li>
<li>Report a bug at <br><a href="https://github.com/SydneyBioX/scMerge/issues">https://​github.com/​SydneyBioX/​scMerge/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>Unpublished work, please do not distribute.</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Yingxin Lin <br><small class="roles"> Author, maintainer </small>  </li>
<li>Kevin Wang <br><small class="roles"> Author </small>  </li>
<li>Sydney Bioinformatics and Biometrics Group <br><small class="roles"> Funder </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Yingxin Lin, Kevin Wang, Sydney Bioinformatics and Biometrics Group.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
