<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the scMerge package • scMerge</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the scMerge package">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scMerge</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scMerge.html">Vignette</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Gallery
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Mouse_Embryonic_Data/Mouse_Embyronic_Data_Processing.html">Mouse Embryonic Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/scMerge">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>An introduction to the scMerge package</h1>
                        <h4 class="author">Yingxin Lin and Kevin Wang</h4>
            
            <h4 class="date">2018-08-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SydneyBioX/scMerge/blob/master/vignettes/scMerge.Rmd"><code>vignettes/scMerge.Rmd</code></a></small>
      <div class="hidden name"><code>scMerge.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The scMerge algorithm allows batch effect removal and normalisation for single cell RNA-Seq data. It comprises of three key components including:</p>
<ol style="list-style-type: decimal">
<li>The identification of stably expressed genes (SEGs) as “negative controls” for estimating unwanted factors;</li>
<li>The construction of pseudo-replicates to estimate the effects of unwanted factors; and</li>
<li>The adjustment of the datasets with unwanted variation using a fastRUVIII model.</li>
</ol>
<p>The purpose of this vignette is to illustrate some uses of <code>scMerge</code> and explain its key components.</p>
</div>
<div id="loading-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-packages-and-data" class="anchor"></a>Loading Packages and Data</h1>
<p>We will load the <code>scMerge</code> package. We designed our package to be consistent with the popular BioConductor’s single cell analysis framework, namely the <code>SingleCellExperiment</code> package and <code>scater</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
  <span class="kw">library</span>(scMerge)
  <span class="kw">library</span>(SingleCellExperiment)
  <span class="kw">library</span>(scater)
})</code></pre></div>
<p>We provided an illustrative mouse embryonic stem cell (mESC) data in our package, as well as a set of pre-computed stably expressed gene (SEG) list to be used as negative control genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## mouse ESC data
<span class="kw">data</span>(<span class="st">"sce_mESC"</span>)
## single-cell stably expressed gene list
<span class="kw">data</span>(<span class="st">"segList_ensemblGeneID"</span>)

sce_mESC
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 24224 704 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(24224): ENSMUSG00000000001 ENSMUSG00000000028 ...</span>
<span class="co">#&gt;   ENSMUSG00000060565 ENSMUSG00000080069</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(704): ola_mES_lif_1_1.counts ola_mES_lif_1_10.counts ...</span>
<span class="co">#&gt;   ola_mES_2i_5_95.counts ola_mES_2i_5_96.counts</span>
<span class="co">#&gt; colData names(2): cellTypes batch</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; spikeNames(0):</span></code></pre></div>
<p>In the mESC data, we pooled data from 5 different batches from three different cell types. Using a PCA plot, we can see that despite strong separation of cell types, there is also a strong separation due to batch effects. This information is stored in the <code>colData</code> of <code>sce_mESC</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(sce_mESC<span class="op">$</span>batch, sce_mESC<span class="op">$</span>cellTypes)
<span class="co">#&gt;         </span>
<span class="co">#&gt;          2i a2i lif</span>
<span class="co">#&gt;   batch1  0   0  81</span>
<span class="co">#&gt;   batch2 82  93  90</span>
<span class="co">#&gt;   batch3 59  66  79</span>
<span class="co">#&gt;   batch4 72   0   0</span>
<span class="co">#&gt;   batch5 82   0   0</span>

scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/checking%20raw%20data-1.png" width="700"></p>
</div>
<div id="illustrating-pseudo-replicates-constructions" class="section level1">
<h1 class="hasAnchor">
<a href="#illustrating-pseudo-replicates-constructions" class="anchor"></a>Illustrating pseudo-replicates constructions</h1>
<p>The first major component of <code>scMerge</code> is to obtain negative controls for our normalisation. In this vignette, we will be using a set of pre-computed SEGs from a single cell mouse data.</p>
<p>The second major component of <code>scMerge</code> is to compute pseudo-replicates for cells so we can perform normalisation. We offer three major ways of computing this pseudo-replicate information:</p>
<ol style="list-style-type: decimal">
<li>Unsupervised clustering, using k-means clustering;</li>
<li>Supervised clustering, using known cell type information; and</li>
<li>Semi-supervised clustering, using partially known cell type information.</li>
</ol>
<div id="unsupervised-scmerge" class="section level2">
<h2 class="hasAnchor">
<a href="#unsupervised-scmerge" class="anchor"></a>Unsupervised <code>scMerge</code>
</h2>
<p>In unsupervised <code>scMerge</code>, we will perform a k-means clustering to obtain pseudo-replicates. This requires the users to supply a <code>kmeansK</code> vector with each element indicating number of clusters in each of the batches. For example, we know the first, forth, and fifth batch contains only one cell type and the second and third batch contains three cell types. Hence, <code>kmeansK = c(1,3,3,1,1)</code> in this case.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(<span class="dt">sce_combine =</span> sce_mESC, 
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_unsupervised"</span>)</code></pre></div>
<p>We now colour construct the PCA plot again on our normalised data. We can observe a much better separation by cell type and less separation by batches.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_unsupervised"</span>)

scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/unsupervised_default_plotting-1.png" width="700"></p>
<!-- ##Selecting 80% of cells -->
<!-- ```{r results='hide',fig.show='hide'} -->
<!-- system.time(sce_mESC <- scMerge(sce_mESC,  -->
<!--                     ctl = segList_ensemblGeneID$mouse$mouse_scSEG, -->
<!--                     kmeansK = c(1,3,3,1,1), -->
<!--                     assay_name = "scMerge_unsupervised_80", -->
<!--                     replicate_prop = 0.8)) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- sce_mESC <- runPCA(sce_mESC, exprs_values = "scMerge_unsupervised_80") -->
<!-- scater::plotPCA(sce_mESC, colour_by="cellTypes",shape_by="batch") -->
<!-- ``` -->
<div id="selecting-all-cells" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-all-cells" class="anchor"></a>Selecting all cells</h3>
<p>By default, <code>scMerge</code> only uses 50% of the cells to perform kmeans clustering. While this is sufficient to perform a satisfactory normalisation in most cases, users can control if they wish all cells be used in the kmeans clustering.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC, 
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_unsupervised_all"</span>,
                    <span class="dt">replicate_prop =</span> <span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_unsupervised_all"</span>)
scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/unsupervised_prop1_plotting-1.png" width="700"></p>
<!-- ##Selecting 20% of cells -->
<!-- ```{r results='hide',fig.show='hide'} -->
<!-- system.time(sce_mESC <- scMerge(sce_mESC,  -->
<!--                     ctl = segList_ensemblGeneID$mouse$mouse_scSEG, -->
<!--                     kmeansK = c(1,3,3,1,1), -->
<!--                     assay_name = "scMerge_unsupervised_50", -->
<!--                     replicate_prop = 0.2)) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- sce_mESC <- runPCA(sce_mESC, exprs_values = "scMerge_unsupervised_50") -->
<!-- scater::plotPCA(sce_mESC, colour_by="cellTypes",shape_by="batch") -->
<!-- ``` -->
</div>
<div id="supervised-scmerge" class="section level3">
<h3 class="hasAnchor">
<a href="#supervised-scmerge" class="anchor"></a>Supervised <code>scMerge</code>
</h3>
<p>If <strong>all</strong> cell type information is available to the user, then it is possible to use this information to create pseudo-replicates. This can be done through the <code>cell_type</code> argument in the <code>scMerge</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC,
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_supervised"</span>,
                    <span class="dt">cell_type =</span> sce_mESC<span class="op">$</span>cellTypes)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_supervised"</span>)
scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>,
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/supervised_plotting-1.png" width="700"></p>
</div>
<div id="semi-supervised-scmerge-i" class="section level3">
<h3 class="hasAnchor">
<a href="#semi-supervised-scmerge-i" class="anchor"></a>Semi-supervised scMerge I</h3>
<p>If the user is only able to access <strong>partial</strong> cell type information, then it is still possible to use this information to create pseudo-replicates. This can be done through the <code>cell_type</code> and <code>cell_type_inc</code> arguments in the <code>scMerge</code> function. <code>cell_type_inc</code> should contain a vector of indices indicating which elements in the <code>cell_type</code> vector should be used to perform semi-supervised scMerge.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC,
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_semisupervised1"</span>,
                    <span class="dt">cell_type =</span> sce_mESC<span class="op">$</span>cellTypes,
                    <span class="dt">cell_type_inc =</span> <span class="kw">which</span>(sce_mESC<span class="op">$</span>cellTypes <span class="op">==</span><span class="st"> "2i"</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_semisupervised1"</span>)
scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/semi_supervised1_plotting-1.png" width="700"></p>
</div>
<div id="semi-supervised-scmerge-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#semi-supervised-scmerge-ii" class="anchor"></a>Semi-supervised scMerge II</h3>
<!-- Perform scMerge using known cell type information to identify mutual nearest cluster -->
<p>There is alternative semi-supervised method to create pseudo-replicates for <code>scMerge</code>. This uses known cell type information to identify mutual nearest clusters and it is achieved via the <code>cell_type</code> and <code>cell_type_match = TRUE</code> options in the <code>scMerge</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC,
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_semisupervised2"</span>,
                    <span class="dt">cell_type =</span> sce_mESC<span class="op">$</span>cellTypes,
                    <span class="dt">cell_type_match =</span> <span class="ot">TRUE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_semisupervised2"</span>)
scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/semi_supervised2_plotting-1.png" width="700"></p>
</div>
</div>
<div id="achieving-fast-computation" class="section level2">
<h2 class="hasAnchor">
<a href="#achieving-fast-computation" class="anchor"></a>Achieving fast computation</h2>
<p>Under most circumstances, <code>scMerge</code> is fast enough to be used on a personal laptop for a moderately large data like the mESC data (24224 features and 704 cells). For example, a basic run of the <code>scMerge</code> algorithm takes roughly 40 seconds.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">t1 =<span class="st"> </span><span class="kw">Sys.time</span>()
sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC, 
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_classical"</span>)
<span class="co">#&gt; Finding HVG...</span>
<span class="co">#&gt; [1] 3082</span>
<span class="co">#&gt; Clustering within each batch...</span>
<span class="co">#&gt; Performing pca...</span>
<span class="co">#&gt; Creating Mutual Nearest Cluster...</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; [1] 6</span>
<span class="co">#&gt; [1] 7</span>
<span class="co">#&gt;   group batch cluster</span>
<span class="co">#&gt; 1     3     2       1</span>
<span class="co">#&gt; 2     3     3       3</span>
<span class="co">#&gt; 3     2     2       2</span>
<span class="co">#&gt; 4     2     3       1</span>
<span class="co">#&gt; 5     1     2       3</span>
<span class="co">#&gt; 6     1     3       2</span>
<span class="co">#&gt; 7     2     1       1</span>
<span class="co">#&gt; 8     1     4       1</span>
<span class="co">#&gt; 9     1     5       1</span>
<span class="co">#&gt; Dimension of the replicates mapping matrix </span>
<span class="co">#&gt; [1] 704 356</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Performing RUV normalisation... This might take a few minutes...</span>
t2 =<span class="st"> </span><span class="kw">Sys.time</span>()
t2 <span class="op">-</span><span class="st"> </span>t1
<span class="co">#&gt; Time difference of 34.43266 secs</span></code></pre></div>
<p>However, we do recognise the difficulties associated with computation when dealing with larger data. To this end, we devised a fast version of <code>scMerge</code>. The major difference between the two versions lies on the noise estimation component, which utilised singular value decomposition (SVD). In order to speed up <code>scMerge</code>, we used the randomised SVD algorithm, which is able to obtain a very accurate approximation of the noise structure in the data by performing a partial SVD. This option is achieved via the option <code>fast_svd = TRUE</code> and <code>rsvd_prop</code>. <code>rsvd_prop</code> is a parameter between 0 and 1, controlling the degree of approximations.</p>
<p>We recommend only use this option when the number of cells is large in your single cell data. The speed advantage we obtain for large signgle cell data is much more dramatic than on a smaller dataset like the example mESC data.</p>
<p>For instance, in a single cell data with 21700 features and 5526 cells, we saw a 5-fold reduction in computational time when <code>fast_svd = TRUE</code> was invoked.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">t1 =<span class="st"> </span><span class="kw">Sys.time</span>()
sce_mESC &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scMerge.html">scMerge</a></span>(sce_mESC, 
                    <span class="dt">ctl =</span> segList_ensemblGeneID<span class="op">$</span>mouse<span class="op">$</span>mouse_scSEG,
                    <span class="dt">kmeansK =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>),
                    <span class="dt">assay_name =</span> <span class="st">"scMerge_fast"</span>, 
                    <span class="dt">fast_svd =</span> <span class="ot">TRUE</span>, 
                    <span class="dt">rsvd_prop =</span> <span class="fl">0.05</span>)
<span class="co">#&gt; Finding HVG...</span>
<span class="co">#&gt; [1] 3082</span>
<span class="co">#&gt; Clustering within each batch...</span>
<span class="co">#&gt; Performing pca...</span>
<span class="co">#&gt; Creating Mutual Nearest Cluster...</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; [1] 6</span>
<span class="co">#&gt; [1] 7</span>
<span class="co">#&gt;   group batch cluster</span>
<span class="co">#&gt; 1     2     2       1</span>
<span class="co">#&gt; 2     2     3       3</span>
<span class="co">#&gt; 3     3     2       2</span>
<span class="co">#&gt; 4     3     3       1</span>
<span class="co">#&gt; 5     1     2       3</span>
<span class="co">#&gt; 6     1     3       2</span>
<span class="co">#&gt; 7     2     1       1</span>
<span class="co">#&gt; 8     1     4       1</span>
<span class="co">#&gt; 9     1     5       1</span>
<span class="co">#&gt; Dimension of the replicates mapping matrix </span>
<span class="co">#&gt; [1] 704 356</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Performing RUV normalisation... This might take a few minutes...</span>
t2 =<span class="st"> </span><span class="kw">Sys.time</span>()
t2 <span class="op">-</span><span class="st"> </span>t1
<span class="co">#&gt; Time difference of 30.70934 secs</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce_mESC &lt;-<span class="st"> </span>scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce_mESC, 
                           <span class="dt">exprs_values =</span> <span class="st">"scMerge_fast"</span>)
scater<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(sce_mESC, 
                <span class="dt">colour_by =</span> <span class="st">"cellTypes"</span>, 
                <span class="dt">shape_by =</span> <span class="st">"batch"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"fast_svd yields similar results to the default version"</span>)</code></pre></div>
<p><img src="scMerge_files/figure-html/computation_rsvd_plotting-1.png" width="700"></p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.1 (2018-07-02)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS High Sierra 10.13.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scater_1.8.3                ggplot2_3.0.0              </span>
<span class="co">#&gt;  [3] SingleCellExperiment_1.2.0  SummarizedExperiment_1.10.1</span>
<span class="co">#&gt;  [5] DelayedArray_0.6.4          BiocParallel_1.14.2        </span>
<span class="co">#&gt;  [7] matrixStats_0.54.0          Biobase_2.40.0             </span>
<span class="co">#&gt;  [9] GenomicRanges_1.32.6        GenomeInfoDb_1.16.0        </span>
<span class="co">#&gt; [11] IRanges_2.14.10             S4Vectors_0.18.3           </span>
<span class="co">#&gt; [13] BiocGenerics_0.26.0         scMerge_0.1.8              </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] bitops_1.0-6             fs_1.2.5                </span>
<span class="co">#&gt;  [3] RColorBrewer_1.1-2       rprojroot_1.3-2         </span>
<span class="co">#&gt;  [5] numDeriv_2016.8-1        tools_3.5.1             </span>
<span class="co">#&gt;  [7] backports_1.1.2          R6_2.2.2                </span>
<span class="co">#&gt;  [9] KernSmooth_2.23-15       vipor_0.4.5             </span>
<span class="co">#&gt; [11] lazyeval_0.2.1           colorspace_1.3-2        </span>
<span class="co">#&gt; [13] withr_2.1.2              tidyselect_0.2.4        </span>
<span class="co">#&gt; [15] gridExtra_2.3            compiler_3.5.1          </span>
<span class="co">#&gt; [17] xml2_1.2.0               desc_1.2.0              </span>
<span class="co">#&gt; [19] labeling_0.3             caTools_1.17.1.1        </span>
<span class="co">#&gt; [21] scales_1.0.0             pkgdown_1.1.0           </span>
<span class="co">#&gt; [23] commonmark_1.5           stringr_1.3.1           </span>
<span class="co">#&gt; [25] digest_0.6.15            rmarkdown_1.10          </span>
<span class="co">#&gt; [27] XVector_0.20.0           pkgconfig_2.0.1         </span>
<span class="co">#&gt; [29] htmltools_0.3.6          ruv_0.9.7               </span>
<span class="co">#&gt; [31] bbmle_1.0.20             limma_3.36.2            </span>
<span class="co">#&gt; [33] rlang_0.2.1              rstudioapi_0.7          </span>
<span class="co">#&gt; [35] shiny_1.1.0              prettydoc_0.2.1         </span>
<span class="co">#&gt; [37] DelayedMatrixStats_1.2.0 bindr_0.1.1             </span>
<span class="co">#&gt; [39] gtools_3.8.1             dplyr_0.7.6             </span>
<span class="co">#&gt; [41] RCurl_1.95-4.11          magrittr_1.5            </span>
<span class="co">#&gt; [43] GenomeInfoDbData_1.1.0   Matrix_1.2-14           </span>
<span class="co">#&gt; [45] Rcpp_0.12.18             ggbeeswarm_0.6.0        </span>
<span class="co">#&gt; [47] munsell_0.5.0            Rhdf5lib_1.2.1          </span>
<span class="co">#&gt; [49] viridis_0.5.1            stringi_1.2.4           </span>
<span class="co">#&gt; [51] yaml_2.2.0               edgeR_3.22.3            </span>
<span class="co">#&gt; [53] MASS_7.3-50              zlibbioc_1.26.0         </span>
<span class="co">#&gt; [55] gplots_3.0.1             rhdf5_2.24.0            </span>
<span class="co">#&gt; [57] plyr_1.8.4               grid_3.5.1              </span>
<span class="co">#&gt; [59] gdata_2.18.0             promises_1.0.1          </span>
<span class="co">#&gt; [61] M3Drop_1.6.0             shinydashboard_0.7.0    </span>
<span class="co">#&gt; [63] crayon_1.3.4             lattice_0.20-35         </span>
<span class="co">#&gt; [65] locfit_1.5-9.1           knitr_1.20              </span>
<span class="co">#&gt; [67] pillar_1.3.0             igraph_1.2.2            </span>
<span class="co">#&gt; [69] rjson_0.2.20             reshape2_1.4.3          </span>
<span class="co">#&gt; [71] codetools_0.2-15         glue_1.3.0              </span>
<span class="co">#&gt; [73] evaluate_0.11            data.table_1.11.4       </span>
<span class="co">#&gt; [75] httpuv_1.4.5             foreach_1.4.4           </span>
<span class="co">#&gt; [77] gtable_0.2.0             purrr_0.2.5             </span>
<span class="co">#&gt; [79] assertthat_0.2.0         rsvd_0.9                </span>
<span class="co">#&gt; [81] mime_0.5                 xtable_1.8-2            </span>
<span class="co">#&gt; [83] roxygen2_6.1.0           later_0.7.3             </span>
<span class="co">#&gt; [85] viridisLite_0.3.0        tibble_1.4.2            </span>
<span class="co">#&gt; [87] iterators_1.0.10         beeswarm_0.2.3          </span>
<span class="co">#&gt; [89] memoise_1.1.0            tximport_1.8.0          </span>
<span class="co">#&gt; [91] bindrcpp_0.2.2           statmod_1.4.30</span></code></pre></div>
<!-- You can enable figure captions by `fig_caption: yes` in YAML: -->
<!--     output: -->
<!--       rmarkdown::html_vignette: -->
<!--         fig_caption: yes -->
<!-- Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**. -- -->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#loading-packages-and-data">Loading Packages and Data</a></li>
      <li>
<a href="#illustrating-pseudo-replicates-constructions">Illustrating pseudo-replicates constructions</a><ul class="nav nav-pills nav-stacked">
<li><a href="#unsupervised-scmerge">Unsupervised <code>scMerge</code></a></li>
      <li><a href="#achieving-fast-computation">Achieving fast computation</a></li>
      </ul>
</li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Yingxin Lin, Kevin Wang, Sydney Bioinformatics and Biometrics Group.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
