<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Merge single-cell RNA-seq data from different batches, experiments, protocols. — scMerge • scMerge</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Merge single-cell RNA-seq data from different batches, experiments, protocols. — scMerge" />

<meta property="og:description" content="Merge single-cell RNA-seq data from different batches, experiments, protocols." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scMerge</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scMerge.html">Vignette</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Case Study
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Mouse_Liver_Data/Mouse_Liver_Data.html">Mouse Liver Data</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SydneyBioX/scMerge">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Merge single-cell RNA-seq data from different batches, experiments, protocols.</h1>
    <small class="dont-index">Source: <a href='https://github.com/SydneyBioX/scMerge/blob/master/R/scMerge.R'><code>R/scMerge.R</code></a></small>
    <div class="hidden name"><code>scMerge.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Merge single-cell RNA-seq data from different batches, experiments, protocols.</p>
    
    </div>

    <pre class="usage"><span class='fu'>scMerge</span>(<span class='no'>sce_combine</span>, <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>kmeansK</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>exprs</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>,
  <span class='kw'>hvg_exprs</span> <span class='kw'>=</span> <span class='st'>"counts"</span>, <span class='kw'>marker</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>marker_list</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ruvK</span> <span class='kw'>=</span> <span class='fl'>20</span>, <span class='kw'>replicate_prop</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>cell_type</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>cell_type_match</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>cell_type_inc</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>fast_svd</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>rsvd_prop</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>dist</span> <span class='kw'>=</span> <span class='st'>"cor"</span>, <span class='kw'>WV</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>WV_marker</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>return_all_RUV</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>assay_name</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sce_combine</th>
      <td><p>A <code>SingleCellExperiment</code> object contains the batch-combined matrix with batch info in colData.</p></td>
    </tr>
    <tr>
      <th>ctl</th>
      <td><p>A chatacter vector of negative control. It should have a non-empty intersection with the rows of sce_combine.</p></td>
    </tr>
    <tr>
      <th>kmeansK</th>
      <td><p>A vector indicates the kmeans's K for each batch. The length of kmeansK needs to be the same as the number of batch.</p></td>
    </tr>
    <tr>
      <th>exprs</th>
      <td><p>A string inciating the name of the assay requiring batch correction in sce_combine, default is logcounts.</p></td>
    </tr>
    <tr>
      <th>hvg_exprs</th>
      <td><p>A string inciating the assay that to be used for highly variable genes identification in sce_combine, default is counts.</p></td>
    </tr>
    <tr>
      <th>marker</th>
      <td><p>An optional vector of markers, to be used in calculation of mutual nearest cluster. If no markers input, highly variable genes will be used instead.</p></td>
    </tr>
    <tr>
      <th>marker_list</th>
      <td><p>An optional list of markers for each batch, which will be used in calculation of mutual nearest cluster.</p></td>
    </tr>
    <tr>
      <th>ruvK</th>
      <td><p>An optional integer/vector indicating the number of unwanted variation factors that are removed, default is 20.</p></td>
    </tr>
    <tr>
      <th>replicate_prop</th>
      <td><p>A number indicating the ratio of cells that are included in pseudo-replicates, ranges from 0 to 1.</p></td>
    </tr>
    <tr>
      <th>cell_type</th>
      <td><p>An optional vector indicating the cell type information for each cell in the batch-combined matrix. If it is <code>NULL</code>, pseudo-replicate procedure will be run to identify cell type.</p></td>
    </tr>
    <tr>
      <th>cell_type_match</th>
      <td><p>An optional logical input for whether to find mutual nearest cluster using cell type information.</p></td>
    </tr>
    <tr>
      <th>cell_type_inc</th>
      <td><p>An optional vector indicating the indices of the cells that will be used to supervise the pseudo-replicate procedure.</p></td>
    </tr>
    <tr>
      <th>fast_svd</th>
      <td><p>If <code>TRUE</code>, randomised singular value decomposition will be used for singular value decomposition calculation. We recommend using this option when the number of cells is large (e.g. &gt; 1000).</p></td>
    </tr>
    <tr>
      <th>rsvd_prop</th>
      <td><p>If <code>fast_svd = TRUE</code>, then rsvd_prop will be used to used to reduce the computational cost of randomised singular value decomposition. We recommend setting this number to less than 0.25 to achieve a balance between numerical accuracy and computational costs.</p></td>
    </tr>
    <tr>
      <th>dist</th>
      <td><p>The distance metrics that are used in the calculation of the mutual nearest cluster, default is Pearson correlation.</p></td>
    </tr>
    <tr>
      <th>WV</th>
      <td><p>A optional vector indicating the wanted variation factor other than cell type info, such as cell stages.</p></td>
    </tr>
    <tr>
      <th>WV_marker</th>
      <td><p>An optional vector indicating the markers of the wanted variation.</p></td>
    </tr>
    <tr>
      <th>return_all_RUV</th>
      <td><p>If <code>FALSE</code>, then only returns a <code>SingleCellExperiment</code> object with original data and one normalised matrix.
Otherwise, the <code>SingleCellExperiment</code> object will contain the original data and one normalised matrix for <code>each</code> ruvK value. In this latter case, assay_name must have the same length as ruvK.</p></td>
    </tr>
    <tr>
      <th>assay_name</th>
      <td><p>The assay name(s) for the adjusted expression matrix(matrices). If <code>return_all_RUV = TRUE</code> assay_name must have the same length as ruvK.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a <code>SingleCellExperiment</code> object with following:</p>
<dt>metadata</dt><dd><p>containing the ruvK vector, ruvK_optimal based on F-score, and the replicate matrix</p></dd>
<dt>assays</dt><dd><p>the original matrices and also the normalised matrices</p></dd>

    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>suppressPackageStartupMessages</span>({
<span class='fu'>library</span>(<span class='no'>SingleCellExperiment</span>)
<span class='fu'>library</span>(<span class='no'>scater</span>)
})
<span class='co'># Loading example data</span>
<span class='fu'>data</span>(<span class='st'>"sce_mESC"</span>)
<span class='co'># Previously computed stably expressed genes</span>
<span class='fu'>data</span>(<span class='st'>"segList_ensemblGeneID"</span>)
<span class='co'># Running an example data with minimal inputs</span>
<span class='no'>sce_mESC</span> <span class='kw'>&lt;-</span> <span class='fu'>scMerge</span>(
                      <span class='kw'>sce_combine</span> <span class='kw'>=</span> <span class='no'>sce_mESC</span>,
                      <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='no'>segList_ensemblGeneID</span>$<span class='no'>mouse</span>$<span class='no'>mouse_scSEG</span>,
                      <span class='kw'>kmeansK</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>1</span>,<span class='fl'>3</span>,<span class='fl'>3</span>,<span class='fl'>1</span>,<span class='fl'>1</span>),
                      <span class='kw'>assay_name</span> <span class='kw'>=</span> <span class='st'>"scMerge"</span>)</div><div class='output co'>#&gt; Finding HVG...
#&gt; [1] 3082
#&gt; Clustering within each batch...
#&gt; Performing pca...
#&gt; Creating Mutual Nearest Cluster...
#&gt; [1] 1
#&gt; [1] 2
#&gt; [1] 3
#&gt; [1] 4
#&gt; [1] 5
#&gt; [1] 6
#&gt; [1] 7</div><div class='img'><img src='scMerge-1.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt;   group batch cluster
#&gt; 1     1     2       1
#&gt; 2     1     3       1
#&gt; 3     2     2       2
#&gt; 4     2     3       2
#&gt; 5     3     2       3
#&gt; 6     3     3       3
#&gt; 7     2     1       1
#&gt; 8     1     4       1
#&gt; 9     1     5       1
#&gt; Dimension of the replicates mapping matrix 
#&gt; [1] 704 356
#&gt; 
#&gt; Performing RUV normalisation... This might take a few minutes... </div><div class='input'><span class='kw pkg'>scater</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/scater/topics/plot_reddim'>plotPCA</a></span>(<span class='no'>sce_mESC</span>, <span class='kw'>colour_by</span> <span class='kw'>=</span> <span class='st'>"cellTypes"</span>, <span class='kw'>shape</span> <span class='kw'>=</span> <span class='st'>"batch"</span>,
                 <span class='kw'>run_args</span> <span class='kw'>=</span> <span class='fu'>list</span>(<span class='kw'>exprs_values</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>))</div><div class='img'><img src='scMerge-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='kw pkg'>scater</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/scater/topics/plot_reddim'>plotPCA</a></span>(<span class='no'>sce_mESC</span>, <span class='kw'>colour_by</span> <span class='kw'>=</span> <span class='st'>"cellTypes"</span>, <span class='kw'>shape</span> <span class='kw'>=</span> <span class='st'>"batch"</span>,
                 <span class='kw'>run_args</span> <span class='kw'>=</span> <span class='fu'>list</span>(<span class='kw'>exprs_values</span> <span class='kw'>=</span> <span class='st'>"scMerge"</span>))</div><div class='img'><img src='scMerge-3.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Yingxin Lin, Kevin Wang

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Yingxin Lin, Kevin Wang, Sydney Bioinformatics and Biometrics Group.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

